### PR TITLE
Do not insert extrapolated submap transforms into global_submap_poses.

### PR DESCRIPTION
**Pull request by ojura**
Related to #602.

### CODE REVIEW

File: cartographer/mapping_2d/sparse_pose_graph.cc [hunk 1]
```diff
@@ -383,19 +384,29 @@ void SparsePoseGraph::AddSubmapFromProto(const transform::Rigid3d& global_pose,
...
+  AddWorkItem([this, submap_id, global_pose_2d,
+               is_global_pose_available]() REQUIRES(mutex_) {
     CHECK_EQ(frozen_trajectories_.count(submap_id.trajectory_id), 1);
     submap_data_.at(submap_id).state = SubmapState::kFinished;
-    optimization_problem_.InsertSubmap(submap_id, global_pose_2d);
+    if (is_global_pose_available) {
+      // Probably incorrect not to do this when a global pose is not available.
+      // Perhaps we should follow GrowSubmapTransformsAsNeeded in that case?
+      // Although we don't have the previous submap available currently, so we
+      // can't compute the transform like in GrowSubmapTransformsAsNeeded...
```
Comment (ojura) [inline cartographer/mapping_2d/sparse_pose_graph.cc]: See comment in code.

**Comment by ojura** [comment]
The biggest problem with this PR is that when deserializing, we can't really emulate what happens with submaps that have extrapolated positions, as noted in that code comment. 

When a new submap is created, `InitializeGlobalSubmapPoses` (ex `GrowSubmapTransformsAsNeeded`) extrapolates a pose for the submap based on the last submap's local pose and [calls `optimization_problem_.AddSubmap` with the extrapolated pose:

https://github.com/googlecartographer/cartographer/blob/a4c0e4754e99e461e2497b05f234839b181fc3cc/cartographer/mapping_2d/sparse_pose_graph.cc#L83-L92

 However, in `AddSubmapFromProto` we don't have access to the previous submap, so we can't emulate what `InitializeGlobalSubmapPoses` does when calling `optimization_problem_.InsertSubmap`.

**Comment by ojura** [comment]
Okay, I reworked this. The rationale is simple: if a submap pose was not in `global_submap_poses` previously, it should not be there after deserialization either.

If you are absolutely certain that this is not important and won't affect anything, we can close this PR and #602.

**Comment by wohe** [comment]
Yes, I think it is not important and we can close this PR.

**Pull request closed without merging**